lib/compress.js

```diff
@@ -2251,7 +2251,8 @@ merge(Compressor.prototype, {
                             dirs.push(node);
                             return make_node(AST_EmptyStatement, node);
                         }
-                        if (node instanceof AST_Defun && hoist_funs) {
+                        if (node instanceof AST_Defun && hoist_funs
+                            && (tt.parent() === self || !compressor.has_directive("use strict"))) {
```

Comment:
I debate with myself a bit over this performance vs. minimal change thing, but I'll take your vote now that you've casted it :wink: